### PR TITLE
Patch reset cavern sg on new stage start

### PR DESCRIPTION
**Pull request by dbhart**
The initial release of sansmic is supposed to mimic exactly the behavior of the old SANSMIC program. The sansmic code was set up to match the behavior stated in the SANSMIC documentation; the old documentation did not, however, match the actual behavior of the code. This patch returns the code to mimic the *behavior* of SANSMIC, and issues a warning that this behavior will be changed in the next release.

### CODE REVIEW

**Comment by dbhart** [comment]
@hgmaure if you don't mind reviewing the changes in this PR since you found the bug, I'd appreciate it!

File: .github/workflows/build-wheels.yml
Comment (dbhart) [inline .github/workflows/build-wheels.yml:1]: This is pulling in build process changes and is not related to the bug fix

File: .gitignore
Comment (dbhart) [inline .gitignore:1]: Not part of the bugfix, this change helps VSCode users not check in private workspace files

File: README.md
Comment (dbhart) [inline README.md:1]: Not part of the bugfix, this provides links to badges on pypi

File: src/ext_modules/libsansmic/model.cpp
Comment (dbhart) [inline src/ext_modules/libsansmic/model.cpp:1]: Related to the bugfix. This moved the assignment to the outside of the for loop (which was unnecessary)

Unrelated to the bugfix, adds more explicit license details

File: src/python/sansmic/io.py
Comment (dbhart) [inline src/python/sansmic/io.py:1]: Updates warnings and does not reset cavern sg on "repeat"; this returns sansmic to previous behavior.

File: src/python/sansmic/model.py
Comment (dbhart) [inline src/python/sansmic/model.py:1]: Bug fix: removes the override-to-0 of the cavern SG on "REPEAT" stages that was in the Stage creation section.

File: tests/test_io.py
Comment (dbhart) [inline tests/test_io.py:1]: Bug fix: verifies that value was not changed when specified as non-zero in a REPEAT section.

**Review by hgmaure — APPROVED** [review]
